### PR TITLE
Native beacon state: v3

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

This PR modifies code inside `state/state-native/v3` in a similar fashion as #10121 did for `v2`.

**Which issues(s) does this PR fix?**

Extracted from #9820

### CODE REVIEW

File: beacon-chain/state/state-native/v3/state_trie.go [hunk 1]
```diff
@@ -67,56 +111,51 @@ func InitializeFromProtoUnsafe(st *ethpb.BeaconStateBellatrix) (*BeaconState, er
 	b.sharedFieldReferences[balances] = stateutil.NewRef(1)
 	b.sharedFieldReferences[inactivityScores] = stateutil.NewRef(1) // New in Altair.
 	b.sharedFieldReferences[historicalRoots] = stateutil.NewRef(1)
-	b.sharedFieldReferences[latestExecutionPayloadHeader] = stateutil.NewRef(1) // New in Bellatrix.
+	b.sharedFieldReferences[latestExecutionPayloadHeader] = stateutil.NewRef(1) // New in Merge.
```
Comment (nisdas) [inline beacon-chain/state/state-native/v3/state_trie.go:114]: Change this back

File: beacon-chain/state/state-native/v3/state_trie.go [hunk 2]
```diff
@@ -194,16 +233,17 @@ func (b *BeaconState) Copy() state.BeaconState {
 // HashTreeRoot of the beacon state retrieves the Merkle root of the trie
 // representation of the beacon state based on the eth2 Simple Serialize specification.
 func (b *BeaconState) HashTreeRoot(ctx context.Context) ([32]byte, error) {
-	_, span := trace.StartSpan(ctx, "BeaconStateBellatrix.HashTreeRoot")
+	_, span := trace.StartSpan(ctx, "BeaconStateMerge.HashTreeRoot")
```
Comment (nisdas) [inline beacon-chain/state/state-native/v3/state_trie.go:236]: change back

File: beacon-chain/state/state-native/v3/beacon_state_mainnet.go [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/stateutil"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/types"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+)
+
+// BeaconState defines a struct containing utilities for the eth2 chain state, defining
+// getters and setters for its respective values and helpful functions such as HashTreeRoot().
+type BeaconState struct {
+	genesisTime                  uint64                        `ssz-gen:"true"`
+	genesisValidatorsRoot        customtypes.Byte32            `ssz-gen:"true" ssz-size:"32"`
```
Comment (rauljordan) [inline beacon-chain/state/state-native/v3/beacon_state_mainnet.go:21]: We tend to follow the naming conventions `Bytes32` in the rest of the repository, so maybe we can align on consistency in a future PR

**Review by rauljordan — APPROVED** [review]

**Comment by rauljordan** [comment]
Needs a codeowner approval from @nisdas

**Review by nisdas — APPROVED** [review]

File: beacon-chain/state/state-native/v3/beacon_state_mainnet.go (hunk 3 above)
Comment (rkapka) [reply beacon-chain/state/state-native/v3/beacon_state_mainnet.go:21]: This will be removed. See https://github.com/prysmaticlabs/prysm/issues/10100

**Pull request merged**